ddp-workspace/projects/ddp-dsm-ui/src/app/sharedLearningUpload/sharedLearningsUpload.component.ts

```diff
@@ -0,0 +1,299 @@
+      .open(ConfirmationModalComponent, {data: {fileName}, width: '500px'});
+
+    activeConfirmationDialog.afterClosed()
+      .pipe(
+        take(1),
+        takeUntil(this.takeUntilSubject$)
+      )
+      .subscribe({
+        next: (deleteOrNot: boolean) => deleteOrNot && this.handleFileDeletion(somaticDocumentId)
+      });
+  }
+
+  /* Initial Load */
+  private loadData(): Observable<any> {
+    this.isLoading = true;
+    return this.tabActivated$.pipe(
+      switchMap(() => this.stateService
+        .getSomaticResultsFiles(this.participantId)
+        .pipe(
+          map((somaticResultsFiles: SomaticResultsFile[]) => this.filterDeletedFiles(somaticResultsFiles)),
+          map((somaticResultsFiles: SomaticResultsFile[]) =>
+            somaticResultsFiles.map((somaticResultsFile: SomaticResultsFile) => this.mapStatusesToFile(somaticResultsFile))),
+          tap((somaticResultsFilesWithStatus: SomaticResultsFileWithStatus[]) =>
+            this.stateService.updateState(somaticResultsFilesWithStatus)),
+          finalize(() => this.isLoading = false))
+      ),
+      takeWhile(() => !(!!this.somaticResultsFilesWithStatus) || !!this.errorLoadingData),
+      catchError((error: any) => this.handleError(error))
+    );
+  }
+
+  /* Mappers */
+  private mapStatusesToFile(somaticResultsFile: SomaticResultsFile): SomaticResultsFileWithStatus {
+    return {
+      ...somaticResultsFile,
+      createdAt: somaticResultsFile.createdAt * 1000,
+      deletedAt: somaticResultsFile.deletedAt * 1000,
+      sendToParticipantStatus: {status: HttpRequestStatusEnum.NONE, message: null},
+      deleteStatus: {status: HttpRequestStatusEnum.NONE, message: null},
+      virusStatus: this.handleAndReturnVirusStatusFor(somaticResultsFile)
+    };
+  }
+
+  /* Filter functions */
+  private filterDeletedFiles(somaticResultsFiles: SomaticResultsFile[]): SomaticResultsFile[] {
+    return somaticResultsFiles.filter((somaticFile: SomaticResultsFile) => !(!!somaticFile.deletedByUserId));
+  }
+
+  /* Virus scanner functions */
+  private handleAndReturnVirusStatusFor({
+                                          isVirusFree,
+                                          deletedAt,
+                                          somaticDocumentId
+                                        }: SomaticResultsFile): SomaticResultsFileVirusStatusEnum {
+    const isFileDeleted = !!deletedAt;
+
+    if (!isVirusFree && isFileDeleted) { // file has been scanned and is infected
+      return SomaticResultsFileVirusStatusEnum.INFECTED;
+    } else if (!isVirusFree && !isFileDeleted) {
+      // file has not been scanned, so it till scan file as well
+      this.scanForVirus(somaticDocumentId);
+      return SomaticResultsFileVirusStatusEnum.SCANNING;
+    } else if (isVirusFree && !isFileDeleted) { // file has already been scanned for viruses and is clean
+      return SomaticResultsFileVirusStatusEnum.CLEAN;
+    } else {
+      // It should not happen, but anyway:
```

Comment:
TODO: isVirusFree && isFileDeleted return SomaticResultsFileVirusStatusEnum.CLEAN;
Not currently needed because deleted items are removed from display.

Reply:
I guess you mean the **else** case. It is just an **else** case, which, as the comment says, is not expected to happen, but it is there assuming that if it happened, the file is infected, but the backend did something wrong with response data :D .

If you mean **else if case** -  **isVirusFree && !isFileDeleted**, it works correctly, as per discussion: 

> If isVirusFree is true and deletedAt equals 0 (falsy for JS) - the file is clean

Or if you literally mean this case -> **isVirusFree && isFileDeleted**, I don't see such a one in the code

**FYI: Deleted items are already removed from the array once the files array riches the virus scanning process. Files are filtered by the **deletedByUserId** field _(Line 152nd)_**

Reply:
Please fix the  isVirusFree && isFileDeleted case.  If we want to show these deleted items, this latent defect will appear.